From 2054ce267f617d27df9b02ab44634dba149a6534 Mon Sep 17 00:00:00 2001
From: Heinz Wrobel <Heinz.Wrobel@nxp.com>
Date: Thu, 25 Feb 2021 11:51:23 +0100
Subject: [PATCH] ls2084abbmini Change StreamID allocation

This is based on a patch submitted by
Jakub Sosnovec <jakub.sosnovec@nxp.com> but implemented in a BB Mini
specific file.

Signed-off-by: Heinz Wrobel <Heinz.Wrobel@nxp.com>
---
 include/configs/ls2084abbmini.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/include/configs/ls2084abbmini.h b/include/configs/ls2084abbmini.h
index 9228cfeb3d..a77101c81b 100644
--- a/include/configs/ls2084abbmini.h
+++ b/include/configs/ls2084abbmini.h
@@ -9,6 +9,16 @@
 
 #include "ls2080a_common.h"
 
+/* For complex PCIe scenarios we need more stream IDs there.
+ * This patch tweaks the stream ID allocation to have more on PCIe
+ * by reducing the IDs for DPAA2. BB Mini doesn't use all DPAA2
+ * features, so this seems like a reasonable compromise.
+ */
+#undef FSL_PEX_STREAM_ID_END
+#define FSL_PEX_STREAM_ID_END          42
+#undef FSL_DPAA2_STREAM_ID_START
+#define FSL_DPAA2_STREAM_ID_START      43
+
 #if !defined(CONFIG_SPL_BUILD)
 #define I2C_VOL_MONITOR_ADDR           0x63
 #define CONFIG_VOL_MONITOR_LTC3882_READ
-- 
2.17.1

